### PR TITLE
Misc fixes

### PR DESCRIPTION
**Pull request by Vtec234**
- I linked the webserver on Zulip and realized it doesn't work.
- A fix for hole hovers -- now `a := ..` in a do-block shows the type of `a` on hover rather than just `Type`. This was because there is a synthetic hole there whose *value* is the type.

### CODE REVIEW

**Review by Kha — COMMENTED** [review]
Thanks for updating the code!

**Comment by Kha** [comment]
>     * This was because there is a synthetic hole there whose _value_ is the type.

The server being affected by that is a bug. Fixed in https://github.com/leanprover/lean4/pull/564.

File: tests/playground/webserver/Webserver.lean [hunk 1]
```diff
@@ -100,28 +100,31 @@ syntax "<" ident ">" child* "</" ident ">" : element
...
       mkNodeToken `LX.text startPos c s } -- }}}
 
+@[combinatorFormatter text] def text.formatter : Formatter := pure ()
+@[combinatorParenthesizer text] def text.parenthesizer : Parenthesizer := pure ()
+
 syntax text         : child
 syntax "{" term "}" : child
 syntax element      : child
 
 syntax element : term
```
[the remark this answers is not in the document]
Comment (Vtec234) [reply tests/playground/webserver/Webserver.lean:114]: Oh wow, I tried 500 and thought that was high, but turns out we scale all the way over 10000! For line 172 `syntax:1023` here also works, but the last line still needs parens even with `syntax:max`.

File: tests/playground/webserver/Webserver.lean [hunk 2]
```diff
@@ -131,43 +134,50 @@ open Webserver
...
     mkNodeToken `pathLiteral startPos c s } -- }}}
 
+@[combinatorFormatter pathLiteral] def pathLiteral.formatter : Formatter := pure ()
+@[combinatorParenthesizer pathLiteral] def pathLiteral.parenthesizer : Parenthesizer := pure ()
+
 declare_syntax_cat pathItem
 syntax pathLiteral : pathItem
 syntax "{" ident "}" : pathItem
-syntax path := pathItem*
+declare_syntax_cat path
```
[the remark this answers is not in the document]
Comment (Vtec234) [reply tests/playground/webserver/Webserver.lean:147]: Yes, `syntax path := pathItem*` doesn't seem to generate a kind needed for `` `(path| $[$pis:pathItem]* ) ← p ``

**Pull request merged**